src/testshade/rs_simplerend.cpp

```diff
@@ -179,29 +179,199 @@ rs_transform_points(OSL::OpaqueExecContextPtr /*ec*/, OSL::ustringhash /*from*/,
     return false;
 }
 
-OSL_RSOP bool
+OSL_RSOP OSL_HOSTDEVICE bool
+rs_texture(OSL::OpaqueExecContextPtr ec, OSL::ustringhash filename,
+           OSL::TextureSystem::TextureHandle* texture_handle,
+           OSL::TextureSystem::Perthread* texture_thread_info,
+           OSL::TextureOpt& options, float s, float t, float dsdx, float dtdx,
+           float dsdy, float dtdy, int nchannels, float* result,
+           float* dresultds, float* dresultdt, OSL::ustringhash* errormessage)
+{
+#ifndef __CUDA_ARCH__
+    auto sg = (OSL::ShaderGlobals*)ec;
+    return sg->renderer->texture(filename, texture_handle, texture_thread_info,
+                                 options, sg, s, t, dsdx, dtdx, dsdy, dtdy,
```

Comment:
Let's not have a customization point call a customization point, instead lets take the default implementation of RendererServices::texture that utilizes a texuresys() and put it here as example for host (or device when OIIO texture sys free functions are fully available).